FlowCrypt/Functionality/Pgp/KeyMethods.swift

```diff
@@ -68,7 +61,7 @@ final class KeyMethods: KeyMethodsType {
         if let byAnyUid = try filter(keys, usable, ({ $0.pgpUserEmailsLowercased.contains(senderEmail) })) {
             return byAnyUid // if any keys match by any uid, use them
         }
-        return try filter(keys, usable) ?? [] // use any usable keys
+        return []
```

Comment:
Choosing keys for signing and for encryption should work differently.

For encryption, this should be left as it was before.

For signing, need to follow the steps I explained, which this doesn't.

Reply:
You mean we should return any usable key even if we can't find keys for filtered condition?(which means logged in user's key contains senderEmail info)
> For encryption, this should be left as it was before.

What's different between you mentioned and current implementation?
> For signing, need to follow the steps I explained, which this doesn't.

Reply:
> You mean we should return any usable key even if we can't find keys for filtered condition?(which means logged in user's key contains senderEmail info)

I said encryption selection should be left unchanged except supplying the method with the senderEmail. But the following changed:

```diff
-        return try filter(keys, usable) ?? [] // use any usable keys
+        return []
```

Yes, for encryption, even when we cannot find key by the right uid, we can use the keys that don't match uid. It won't cause much trouble for encryption, but it does for signature verification, and that's why we're treating them differently. We can be more lenient for encryption.

> What's different between you mentioned and current implementation?

I've re-read the implementation for signing keys, and I think it will be ok. It's implemented differently, but I think it's the same. So just need to fix the selection for encryption, back to what it was.

------------------

You can include a variation of `// Yes, for encryption, even when we cannot find key by the right uid, we can use the keys that don't match uid. It won't cause much trouble for encryption, but it does for signature verification, and that's why we're treating them differently. We can be more lenient for encryption.` to make it clearer why we treat the two differently.

Reply:
Ok, got it